### ISSUE TITLE
Meeting 18/2/2015

### ISSUE THREAD

**Issue by Moorkopsoesje** [StevenReitsma/kaggle-national-data-science-bowl#8]
- We are **124th** at the moment. The other teams are all worse than us, so they are no competition.
- Tom and Luc have looked at Steven's code, reused what was reusable and removed what wasn't useful. They wrote k-means but need to look at how many iterations (and you have normal k-means and just call 'fit', and you have minibatches k-means, so work out differences). They still need to use Guido's patches. (they will probably get a matrix, reshape to vector, centroid vector back to matrix again)
- Guido still needs to write the patches to a text-file. Steven suggests a HDF5 file (binary, easy to read). Guido will also write something that loads the patches.
  It doesn't costs a lot of memory (3GB perhaps).
  Guido has looked at S' code, in Matlab, but he wrote his own. It reads all images, then in two ways makes the images square. Stretch and resize. Then it makes patches from the images, e.g. 10 per image. It doesn't make double patches. All possible coordinates are put in a list, and then it takes patches from it. Patches are pieces of the images, so used as a feature. They overlap, but then you can train over the patches.
  There are two lists: 1 list are labels and file names, indexes are the same as the other list, which is a list of lists of patches (which picture (index), width, height, and then all patches). The labels don't matter for unsupervised, so you pick the second list and flatten it. Guido will also make a list with all possible patches, so it's easy to use in training.
- Is the test set distributed the same as the training set? Usually it's quite comparable, so perhaps use probabilities. But may be a risk. Also, how exactly will we use this?
  Test set can be used for unsupervised learning, not for supervised, so you can also use this during training.
- Robbert and Inez took a RBM implementation for digits, and still need to use it for the patches. But this is supervised, so still make it unsupervised, look at other implementations that use other library than just bernoulliRBM. [Geoffrey Hinton paper about RBM](http://www.cs.toronto.edu/~hinton/absps/guideTR.pdf), guide for RBMs. Check Coates to see how to use weights and biases to get 8x8 feature representation.
- Steven tried something with Convolutional neural networks, Torch is a neural network library. It's not in Python, but in Lua. Syntax is nice, but nothing is readily implemented, so command line bash needs to be called. For everything a library is needed. Took a lot of time, and is crap. But easily extendable. 
  Then looked at CXX net, written in C++. Config files can be made, in Python preprocessing and just throw it in binary file with config file, to define network. That's used to obtain current score.
  First 1.40 with this method and current training set. Then for all images in training set also used rotations 90, 180 and 270 degrees, to obtain current score. Other rotations are more difficult, need to fill the squares, but probably worth a try.

To-do for next Saturday:
- [x] @robbertvdg needs to become a Slackertje. 
- [x] [RU] looks like we are Russian, so @StevenReitsma is going to change it to [ru.nl] or remove it.
- [x] @Guardianofnature & @Lucivius k-means test on patches, check output if 8x8 feature rep
- [x] @Rahazan normalize patches and write to file
- [x] @robbertvdg & @Moorkopsoesje study supervised and unsupervised RBM, check differences in implementation and look up how to get weights and biases to make feature rep
- [x] @StevenReitsma invites Maaike (coach) to Slack #accomplishments and #general
- [x] First goal: top 25%
